core/modules/util/ThreadPool.cc

```diff
@@ -81,6 +83,11 @@ bool PoolEventThread::leavePool(Command::Ptr const& cmd) {
     // This thread will stop accepting commands
     _loop = false;
     LOGS(_log, LOG_LVL_DEBUG, "PoolEventThread::leavePool " << this);
+    if (_threadPool->atMaxThreadPoolCount()) {
+        LOGS(_log, LOG_LVL_WARN, "A new thread cannot be created immediately as there are too many threads");
+        return false;
+    }
+
```

Comment:
This test is counterintuitive as it's done inside method which I think is supposed to delete the thread. Or, I'm mistaken?

My second comment is that. the Doxygen comment before tis method says:
```
// @return false if a different command is running than cmd.
```
The `if` block returns `false`. Is documentation consistent with the behavior of the method?